test/common/alerts_firing.go

```diff
@@ -61,8 +92,93 @@ func (e *alertsFiringError) isValid() bool {
 	return !e.deadMansSwitchFiring || len(e.alertsFiring) != 0 || len(e.alertsPending) != 0
 }
 
-// TestIntegreatlyAlertsFiring reports any firing or pending alerts
+// reports any alerts that are firing only
 func TestIntegreatlyAlertsFiring(t *testing.T, ctx *TestingContext) {
+	//fail immediately if one or more alerts have fired
+	if err := getFiringAlerts(t, ctx); err != nil {
+		podLogs(t, ctx)
+		t.Fatal(err)
+	}
+
+}
+func getFiringAlerts(t *testing.T, ctx *TestingContext) error {
+	output, err := execToPod("curl localhost:9090/api/v1/alerts",
+		"prometheus-application-monitoring-0",
+		MonitoringOperatorNamespace,
+		"prometheus",
+		ctx)
+	if err != nil {
+		return fmt.Errorf("failed to exec to prometheus pod: %w", err)
+	}
+
+	// get all found rules from the prometheus api
+	var promApiCallOutput prometheusAPIResponse
+	err = json.Unmarshal([]byte(output), &promApiCallOutput)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal json: %w", err)
+	}
+	var alertsResult prometheusv1.AlertsResult
+	err = json.Unmarshal(promApiCallOutput.Data, &alertsResult)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal json: %w", err)
+	}
+
+	// create a custom alerts error to keep track of all firing alerts
+	alertsError := &alertsFiringError{
+		alertsFiring:         []alertTestMetadata{},
+		deadMansSwitchFiring: true,
+	}
+
+	// // check if any alerts other than DeadMansSwitch are firing
+	for _, alert := range alertsResult.Alerts {
+		alertName := string(alert.Labels["alertname"])
+		alertMetadata := alertTestMetadata{
+			alertName: alertName,
+			podName:   string(alert.Labels["pod"]),
+			namespace: string(alert.Labels["namespace"]),
+		}
+
+		// check if dead mans switch is firing
+		if alertName == deadMansSwitch && alert.State != prometheusv1.AlertStateFiring {
+			alertsError.deadMansSwitchFiring = false
+		}
+		// check for firing alerts
+		if alertName != deadMansSwitch {
+			if alert.State == prometheusv1.AlertStateFiring {
+				alertsError.alertsFiring = append(alertsError.alertsFiring, alertMetadata)
+
+			}
+
+		}
+	}
+
+	if alertsError.isValid() {
+		return alertsError
+
+	}
+	return nil
+}
+
+// Makes a api call a to get all pods in the rhmi namespaces
+func podLogs(t *testing.T, ctx *TestingContext) {
+	pods := &corev1.PodList{}
+
+	for _, namespaces := range podNamespaces {
+		err := ctx.Client.List(goctx.TODO(), pods, &k8sclient.ListOptions{Namespace: namespaces})
+		if err != nil {
+			t.Error("Error getting namespaces:", err)
+		}
+		for _, podlogs := range pods.Items {
+			logrus.Infoln("Podname :", podlogs.Name)
+			logrus.Infoln("Namespace :", podlogs.Namespace)
+			logrus.Infoln("Status :", podlogs.Status)
+
+		}
+	}
+}
```

Comment:
Just looking at the output of this function, I was wondering if there was any way to make it slightly more readable when sifting through the logs if the test fails. I made a few modifications here: 

```go
func podLogs(t *testing.T, ctx *TestingContext) {
	pods := &corev1.PodList{}

	for _, namespace := range podNamespaces {
		err := ctx.Client.List(goctx.TODO(), pods, &k8sclient.ListOptions{Namespace: namespace})
		if err != nil {
			t.Error("Error getting namespaces:", err)
		}
		t.Log("Namespace:", namespace)
		for _, pod := range pods.Items {
			t.Logf("\tPod name: %s, Status: %s", pod.Name, pod.Status.Phase)
		}
	}
}
```
This will group the pods by namespace and also show the status `Phase` for each pod which makes it a bit more readable too:
```
        --- FAIL: TestIntegreatly/Integreatly_Happy_Path_Tests/C01_-_Verify_Alerts_are_not_firing_apart_from_DeadMansSwitch (3.92s)
            alerts_firing.go:172: Namespace: redhat-rhmi-operator
            alerts_firing.go:174: 	Pod name: codeready-postgres-rhmi-64b69759df-wmtjq, Status: Running
            alerts_firing.go:174: 	Pod name: fuse-postgres-rhmi-5f5c66d69c-69qsl, Status: Running
            alerts_firing.go:174: 	Pod name: rhsso-postgres-rhmi-789fb876c9-zvp55, Status: Running
            alerts_firing.go:174: 	Pod name: rhssouser-postgres-rhmi-7fc45c6c89-2l9m5, Status: Running
            alerts_firing.go:174: 	Pod name: standard-authservice-postgresql-7885d8c879-zzwrq, Status: Running
            alerts_firing.go:174: 	Pod name: threescale-backend-redis-rhmi-744b798bf8-bgnb7, Status: Running
            alerts_firing.go:174: 	Pod name: threescale-postgres-rhmi-75d4f664d7-7tt67, Status: Running
            alerts_firing.go:174: 	Pod name: threescale-redis-rhmi-5b4648c88b-2lrb4, Status: Running
            alerts_firing.go:174: 	Pod name: ups-postgres-rhmi-85d4f6b745-t6kqn, Status: Running
            alerts_firing.go:172: Namespace: redhat-rhmi-middleware-monitoring-operator
            alerts_firing.go:174: 	Pod name: alertmanager-application-monitoring-0, Status: Running
            alerts_firing.go:174: 	Pod name: application-monitoring-operator-7ff7f8dc89-rkrmz, Status: Running
            alerts_firing.go:174: 	Pod name: grafana-deployment-6fc76bfccb-8gkjz, Status: Running
            alerts_firing.go:174: 	Pod name: grafana-operator-66497b6fc6-rj2tq, Status: Running
            alerts_firing.go:174: 	Pod name: prometheus-application-monitoring-0, Status: Running
            alerts_firing.go:174: 	Pod name: prometheus-operator-5b4c4dff88-7r7dd, Status: Running
            alerts_firing.go:174: 	Pod name: rhmi-registry-cs-4j84v, Status: Running
```

Reply:
Yeah that looks good i wasnt sure if there was other info in the status that might have been useful but i think your right with have just the phase

Reply:
Yes both are being called in common/tests.go. Yeah it would make sense to have it being called in both

Reply:
Holy moly my bad i havent commited the tests.go with the new changes